src/k8s/pkg/k8sd/database/cluster_configs.go

```diff
@@ -0,0 +1,112 @@
+	K8sDqliteCert string `mapstructure:"k8s-dqlite-crt,omitempty"`
+	// K8sDqliteKey represents the Kubernetes Dqlite private key.
+	K8sDqliteKey string `mapstructure:"k8s-dqlite-key,omitempty"`
+	// K8sClusterCIDR represents the Kubernetes Cluster CIDR.
+	K8sClusterCIDR string `mapstructure:"k8s-cluster-cidr,omitempty"`
+	// KubeletClusterDNS represents the DNS address for the Kubelet in the cluster.
+	KubeletClusterDNS string `mapstructure:"kubelet-cluster-dns,omitempty"`
+	// KubeletClusterDomain represents the domain for the Kubelet in the cluster.
+	KubeletClusterDomain string `mapstructure:"kubelet-cluster-domain,omitempty"`
+	// KubeletCloudProvider represents the cloud provider for the Kubelet.
+	KubeletCloudProvider string `mapstructure:"kubelet-cloud-provider,omitempty"`
+	// ApiserverRBAC defines if RBAC (Role-Based Access Control) is enabled for this cluster.
+	ApiserverRBAC bool `mapstructure:"apiserver-rbac,omitempty"`
+	// ApiserverDatastore represents the data store configuration for the API server.
+	// "k8s-dqlite" by default, "etcd" if using a custom datastore.
+	ApiserverDatastore string `mapstructure:"apiserver-datastore,omitempty"`
+	// ApiserverEtcdURL represents the URL for the external etcd service used by the API server.
+	ApiserverEtcdURL string `mapstructure:"apiserver-etcd-url,omitempty"`
+	// ApiserverEtcdCA represents the Certificate Authority for external etcd used by the API server.
+	ApiserverEtcdCA string `mapstructure:"apiserver-etcd-ca,omitempty"`
+	// ApiserverEtcdClientCrt represents the client certificate for external etcd used by the API server.
+	ApiserverEtcdClientCrt string `mapstructure:"apiserver-etcd-client-crt,omitempty"`
+	// ApiserverEtcdClientKey represents the client private key for external etcd used by the API server.
+	ApiserverEtcdClientKey string `mapstructure:"apiserver-etcd-client-key,omitempty"`
+}
+
+// UpsertClusterConfigEntry inserts or updates a single cluster config entry.
+func UpsertClusterConfigEntry(ctx context.Context, tx *sql.Tx, key string, value any) error {
+	upsertTxStmt, err := cluster.Stmt(tx, clusterConfigsStmts["upsert-config"])
+	if err != nil {
+		return fmt.Errorf("failed to prepare upsert statement: %w", err)
+	}
+
+	if _, err := upsertTxStmt.ExecContext(ctx, key, value); err != nil {
+		return fmt.Errorf("upsert query (write %s->%s) failed: %w", key, value, err)
+	}
+
+	return nil
+}
+
+// UpsertConfigs inserts or updates a cluster config instance.
+// Members that are not set will be emptied.
+func UpsertClusterConfigs(ctx context.Context, tx *sql.Tx, clusterConfig ClusterConfig) error {
+	configMap := make(map[string]any)
+	if err := mapstructure.Decode(clusterConfig, &configMap); err != nil {
+		return fmt.Errorf("failed to decode cluster config: %w", err)
+	}
+
+	for key, value := range configMap {
+		if err := UpsertClusterConfigEntry(ctx, tx, key, value); err != nil {
+			return fmt.Errorf("upsert query (write %s->%s) failed: %w", key, value, err)
+		}
+	}
```

Comment:
This works, but requires N queries for N updates. This should not be a problem now, but something to keep in mind.

Reply:
True, I added a TODO for it.